akka-remote/src/test/scala/akka/remote/artery/compress/CompressionIntegrationSpec.scala

```diff
@@ -280,35 +284,57 @@ class CompressionIntegrationSpec extends ArteryMultiNodeSpec(CompressionIntegrat
       expectMsgType[ActorIdentity].ref.get
     }
 
-    val maxTableVersions = 130 // so table version wraps around at least once
-    var lastVersion = 0
+    var seenTableVersions = List.empty[Int]
+    // iterate from 2, since our assertion wants the locally created actor to be included in the table
+    // which will only happen in the 2nd advertisement the earliest.
+    val upToNTablesAcceptedAfterWrap = 6
+    var remainingExpectedTableVersions = (Iterator.from(2).take(126) ++ Iterator.from(0).take(upToNTablesAcceptedAfterWrap + 1)).toList
+
+    // so table version wraps around at least once
     var lastTable: CompressionTable[ActorRef] = null
     var allRefs: List[ActorRef] = Nil
 
-    for (iteration ← 1 to maxTableVersions) {
-      val echoWrap = createAndIdentify(iteration) // create a different actor for every iteration
-      allRefs ::= echoWrap
+    within(3.minutes) {
+      var iteration = 0
+      while (remainingExpectedTableVersions.nonEmpty) {
+        iteration += 1
+        val echoWrap = createAndIdentify(iteration) // create a different actor for every iteration
+        allRefs ::= echoWrap
+
+        // cause echo to become a heavy hitter
+        (1 to messagesToExchange).foreach { i ⇒ echoWrap ! TestMessage("hello") }
+        receiveN(messagesToExchange) // the replies
+
+        var currentTable: CompressionTable[ActorRef] = null
+        receivedActorRefCompressionTableProbe.awaitAssert {
```

Comment:
what is the remaining timeout for this probe? how long should it retry the awaitAssert?

Reply:
Good catch, it seems it reaches into the 3 minutes timeout from the outer within. I'll put a specific max here

Reply:
Ah it'd be bounded by the expectMsgType being 10 seconds internally anyway, but I'll make it explcitic